Bio/SeqIO/SnapGeneIO.py

```diff
@@ -109,10 +109,15 @@ def _parse_cookie_packet(length, data, record):
         raise ValueError("The file is not a valid SnapGene file")
 
 
-def _parse_location(rangespec, strand, record):
+def _parse_location(rangespec, strand, record, is_primer=False):
     start, end = (int(x) for x in rangespec.split("-"))
     # Account for SnapGene's 1-based coordinates
     start = start - 1
+    if is_primer:
+        # Primers' coordinates in SnapGene files are shifted by -1
+        # for some reasons
+        start += 1
+        end += 1
```

Comment:
This off-by-one in the primers only really does feel like a bug in SnapGene. I take it you have reproduced it with other input files?

Reply:
Yes, and I have tested with two different versions of SnapGene (5.3.2 and the latest 6.2): both versions write primers with shifted coordinates.

Whatever the reason for this shift, it is clearly done intentionally, since SnapGene shifts the coordinates _back_ when reading a file in order to display the correct coordinates.

My wild guess is that there must have been a bug in an early version of SnapGene, that caused the writer to shift all primer coordinates by -1. When they realised that, SnapGene developers were faced with a choice:

* fixing the writer so that it would now write correct coordinates; but then, the _reader_ would have to deal with the fix (“has this file that I am reading been generated by a version that suffered from the coordinates shifting bug? then I need to shift coordinates back; otherwise I can use the coordinates from the file as they are”);
* leave the bug in the writer and amend the parser to “cancel” the effects of the bug by systematically shifting coordinates of primers, no matter the version of SnapGene used to generate the file.

And they went for the second solution.

I have no idea if this is really what happened, but that’s the only explanation I can think of that kind of makes sense.

Reply:
Strange, but I agree that would explain it.